go.mod

```diff
@@ -1,5 +1,16 @@
 module github.com/jonnenauha/prometheus_varnish_exporter
 
-go 1.12
+go 1.22
```

Comment:
Why would go 1.22 be required? I get the version bump if deps require it, but im not using packages that need new APIs from later go releases.

Prom seems to require go 1.21 so we should target that. https://github.com/prometheus/client_golang/blob/main/go.mod#L3

Dispite what the go.mod says here, whatever go version you have will be used to build the binary. Its not like if you have go 1.23 installed and build it, it's not going to be be like you built it with go 1.12. go mod tells you the minimum version you can use to build the code. So if you are using features released in later go versions you can enforce it here.

But I agree bumping the version up to 1.21 is sensible. It can probably enable some compiler features versus 1.12.

Reply:
One more note, the travis ci go version update is ofc importatnt as it pushes the offiicial releases to github. So that is great, I was more talking about the go.mod changes here.

Reply:
EDIT: Actually, since 1.23 has been released this is no longer correct. I wouldn't set 1.21 here since it is no [longer supported](https://go.dev/doc/devel/release#policy):
> Each major Go release is supported until there are two newer major releases.